api/src/main/java/org/commonjava/indy/conf/IndyLoggingConfig.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (C) 2011-2018 Red Hat, Inc. (https://github.com/Commonjava/indy)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.conf;
+
+import org.commonjava.web.config.ConfigurationException;
+import org.commonjava.web.config.annotation.ConfigName;
+import org.commonjava.web.config.annotation.SectionName;
+import org.commonjava.web.config.section.MapSectionListener;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.io.File;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by yma on 2019/3/19.
+ */
+
+@ApplicationScoped
+@SectionName( IndyLoggingConfig.SECTION_NAME )
+public class IndyLoggingConfig
+        extends MapSectionListener
+        implements IndyConfigInfo
+{
```

Comment:
You don't need both MapSectionListener and IndyConfigInfo. You just need one of them. I.e. either override the parameter method or use annotations.

Reply:
When I use @ ConfigName annotations, MDCManager could not get the config name's value from setter, so I override the parameter to force the value setting, and the values were gotten. I suppose that's related with CDI load seq, do you have some good suggestions?

Reply:
I think we're using IndyConfigInfo as a marker interface primarily, and most commonly we're using bean-style assignment of configurations for those classes. I just checked, though, and for map-style configuration sections we do use both MapSectionListener (to present a distinct way of consuming the config info, instead of as a bean), and IndyConfigInfo (to allow injecting an Instance...well, instance...that contains all of our configuration classes.